src/Nethermind/Nethermind.Logging.NLog/NLogLogger.cs

```diff
@@ -45,27 +49,112 @@ public NLogLogger(string loggerName = null)
 
         public void Info(string text)
         {
-            _logger.Info(text);
+            if (IsInfo)
+                Log(LogLevel.Info, text);
         }
 
         public void Warn(string text)
         {
-            _logger.Warn(text);
+            if (IsWarn)
+                Log(LogLevel.Warn, text);
         }
 
         public void Debug(string text)
         {
-            _logger.Debug(text);
+            if (IsDebug)
+                Log(LogLevel.Debug, text);
         }
 
         public void Trace(string text)
         {
-            _logger.Trace(text);
+            if (IsTrace)
+                Log(LogLevel.Trace, text);
+        }
+
+        public void Error(string text, Exception? ex = null)
+        {
+            if (IsError)
+                Log(LogLevel.Error, text, ex);
+        }
+
+        private void Log(LogLevel level, string text, Exception? ex = null)
+        {
+            _workItems.Enqueue(new(level, text, ex));
+
+            // Set working if it wasn't (via atomic Interlocked).
+            if (Interlocked.CompareExchange(ref _doingWork, 1, 0) == 0)
+            {
+                // Wasn't working, schedule.
+                ThreadPool.UnsafeQueueUserWorkItem(this, preferLocal: false);
+            }
+        }
+
+        void IThreadPoolWorkItem.Execute()
```

Comment:
But why?

Reply:
Offload all the Is enabled, are you enabled, iterate registered loggers etc; just to make logging fire and forget on the thread as much as possible, since one thread is very very dominated by block production and anything that can happen elsewhere should

Reply:
Reverted

Reply:
Sure but would queuing it on another thread be similar in heaviness as actually letting nlog do that? Doesn't nlog do that under the hood potentially?

Reply:
Nlog splits at the sources, not at top stack. But I should look into it a bit more and not add it in a fast patch